lib/src/widgets/country_selector/country_selector_navigator.dart

```diff
@@ -428,6 +428,11 @@ class DraggableModalBottomSheetNavigator extends CountrySelectorNavigator {
         maxChildSize: maxChildSize,
         expand: false,
         builder: (context, scrollController) {
+          // hide the keyboard when scrolling the list, so it doesn't overlap the list
+          scrollController.addListener(() {
```

Comment:
this is a memory leak

Reply:
Can you help fixing this issue? I am curious to know how you would do it. If it was a normal stateless widget, I could easily convert it to a stateful one and fix it, but given your setup with factories, I am not familiar with how.

Reply:
Maybe something like this would do:

```
final controller  = new ScrollController();
controller.addListener(...);

final result = await showModalBottomSheet(
  // ...
  DraggableScrollableSheet(
     controller: scrollController
   )
)

scrollController.dispose();   // will be called on close
```

Reply:
@cedvdb here from this other issue: https://github.com/cedvdb/phone_form_field/pull/162


@cedvdb How would you dispose the scrollController? Unfamiliar with using factory for widgets like you do, so would appreciate your feedback. One idea (not tested, and not sure it would work tbh) would be like below, however, it would require removing the const constructor, which isn't ideal either I guess.

Are you sure the builder does not handle the dispose automatically given it's provided by the builder? We haven't seen any issues at least, running this in production for many months now.

```diff
+++ b/lib/src/widgets/country_selector/country_selector_navigator.dart
@@ -372,12 +372,14 @@ class DraggableModalBottomSheetNavigator extends CountrySelectorNavigator {
   final double minChildSize;
   final double maxChildSize;
   final BorderRadiusGeometry? borderRadius;
+  late ScrollController? scrollController;

-  const DraggableModalBottomSheetNavigator._({
+  DraggableModalBottomSheetNavigator._({
     this.initialChildSize = 0.7,
     this.minChildSize = 0.25,
     this.maxChildSize = 0.85,
     this.borderRadius,
+    this.scrollController,
     List<IsoCode>? countries,
     List<IsoCode>? favorites,
     bool addSeparator = true,
@@ -408,7 +410,14 @@ class DraggableModalBottomSheetNavigator extends CountrySelectorNavigator {
           searchBoxIconColor: searchBoxIconColor,
           scrollPhysics: scrollPhysics,
           flagSize: flagSize,
-        );
+        ) {
+    scrollController ??= ScrollController();
+  }
+
+  // Override dispose method to dispose of the ScrollController
+  void dispose() {
+    scrollController?.dispose();
+  }

   @override
   Future<Country?> navigate(BuildContext context) {
@@ -427,9 +436,9 @@ class DraggableModalBottomSheetNavigator extends CountrySelectorNavigator {
         minChildSize: minChildSize,
         maxChildSize: maxChildSize,
         expand: false,
-        builder: (context, scrollController) {
+        builder: (context, _) {
           // hide the keyboard when scrolling the list, so it doesn't overlap the list
-          scrollController.addListener(() {
+          scrollController?.addListener(() {
             FocusManager.instance.primaryFocus?.unfocus();
           });

Reply:
> Are you sure the builder does not handle the dispose automatically given it's provided by the builder

This could be the case. However there might still be an issue where the builder rebuilds and therefor a listener is added on every rebuild. That issue is still there in that last comment.

The addListener would have to move somewhere above.

Alternatively, I proposed :

```dart
 @override
  Future<Country?> navigate(BuildContext context) {
      final scrollController = DraggableScrollableController();
      scrollController.addListener(...)
      final result =  await showModalBottomSheet<Country>(
      context: context,
      builder: (_) => DraggableScrollableSheet(
        controller: scrollController,        
        builder: (context, _) { }
      );
      scrollController.removeListener(...);  // this will be called when the bottom sheet closes
      return result;
```

Reply:
I'm sorry, can you tap into the branch and attempt the changes you suggest?

Reply:
I think it would resolve it much quicker

Reply:
@cedvdb finally - it's been fixed ;)